components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -237,107 +246,121 @@ public void registerOAuthApplicationData(OAuthConsumerAppDTO application) throws
-                        app.setTokenType(application.getTokenType());
-                    } else {
-                        app.setTokenType(getDefaultTokenType());
+                        app.setBackChannelLogoutUrl(application.getBackChannelLogoutUrl());
+                        app.setFrontchannelLogoutUrl(application.getFrontchannelLogoutUrl());
+                        if (application.getTokenType() != null) {
+                            app.setTokenType(application.getTokenType());
+                        } else {
+                            app.setTokenType(getDefaultTokenType());
+                        }
+                        app.setBypassClientCredentials(application.isBypassClientCredentials());
+                        app.setRenewRefreshTokenEnabled(application.getRenewRefreshTokenEnabled());
+                        validateBindingType(application.getTokenBindingType());
+                        app.setTokenBindingType(application.getTokenBindingType());
+                        app.setTokenBindingValidationEnabled(application.isTokenBindingValidationEnabled());
+                        app.setTokenRevocationWithIDPSessionTerminationEnabled(
+                                application.isTokenRevocationWithIDPSessionTerminationEnabled());
                     }
-                    app.setBypassClientCredentials(application.isBypassClientCredentials());
-                    app.setRenewRefreshTokenEnabled(application.getRenewRefreshTokenEnabled());
-                    validateBindingType(application.getTokenBindingType());
-                    app.setTokenBindingType(application.getTokenBindingType());
-                    app.setTokenBindingValidationEnabled(application.isTokenBindingValidationEnabled());
-                    app.setTokenRevocationWithIDPSessionTerminationEnabled(
-                            application.isTokenRevocationWithIDPSessionTerminationEnabled());
-                }
-                dao.addOAuthApplication(app);
-                AppInfoCache.getInstance().addToCache(app.getOauthConsumerKey(), app);
-                if (LOG.isDebugEnabled()) {
-                    LOG.debug("Oauth Application registration success : " + application.getApplicationName() + " in " +
-                            "tenant domain: " + tenantDomain);
+                    dao.addOAuthApplication(app);
+                    AppInfoCache.getInstance().addToCache(app.getOauthConsumerKey(), app);
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Oauth Application registration success : " + application.getApplicationName() +
+                                " in " +
+                                "tenant domain: " + tenantDomain);
+                    }
+                } else {
+                    String message = "No application details in the request. Failed to register OAuth App.";
+                    if (LOG.isDebugEnabled()) {
```

Comment:
we don't need a log.isDebugEnabled check here

Reply:
resolved by 57676650f4bb2e0fb744eae58ff5aebe4aa65b2f